index.js

```diff
@@ -24,74 +24,86 @@ function computeTextColor(uicolors) {
 export const getNamedSwatches = (image, callback) => {
   if (Platform.OS === 'android') {
     return RNPalette.getNamedSwatches(image, callback);
-  } else {
-    callback("Not supported");
   }
-}
-
+  callback('Not supported');
+};
 
+// eslint-disable-next-line consistent-return
 export const getAllSwatches = (options, image, callback) => {
   if (Platform.OS === 'android') {
-    return RNPalette.getAllSwatches(image, callback);
+    return RNPalette.getAllSwatches(image, (err, res) => {
+      if (err) {
+        return callback(err);
+      }
+      res.sort((a, b) => b.population - a.population);
+      const splitColor = res[0].color.split(',');
+      const r = Number.parseFloat(splitColor[0].split('rgba(')[1]);
+      const g = Number.parseFloat(splitColor[1]);
+      const b = Number.parseFloat(splitColor[2]);
+      res[0].hex = `#${toHex(r)}${toHex(g)}${toHex(b)}${toHex(0.6)}`;
+      return callback(err, res);
+    });
   }
-  if (options.hasOwnProperty('threshold')) {
+
+  if (options.threshold) {
     Threshold = options.threshold;
   }
 
-  if (options.hasOwnProperty('quality')) {
-    quality = options.quality
+  if (options.quality) {
+    quality = options.quality;
   }
 
   RNPalette.getColors(image, iOSOptions[quality], (err, res) => {
     if (err) {
       callback(err);
     } else {
-      var props = Object.keys(res);
+      const props = Object.keys(res);
       // convert all dominant to hex
-      var maxResults = 16;
-      var domColors = [];
-      var swatches = [];
+      const maxResults = 16;
+      // const domColors = [];
+      const swatches = [];
       /*           colors.complementColors = []; */
-      for (var i = 0 ; i< props.length && i < maxResults ; i++) {
-        var UIColor = props[i];
-        var uicolors = UIColor.split(' ');
-        var colorSpace = uicolors.shift();
+      for (let i = 0; i < props.length && i < maxResults; i++) {
+        const UIColor = props[i];
+        const uicolors = UIColor.split(' ');
+        uicolors.shift();
 
-        var textColor = computeTextColor(uicolors);
+        const textColor = computeTextColor(uicolors);
 
-        var r = uicolors[0];
-        var g = uicolors[1];
-        var b = uicolors[2];
-        var a = uicolors[3];
-        var hex = '#' + toHex(r*255) + toHex(g*255) +toHex(b*255) + toHex(a*255);
-        var rgba = `rgba(${(r*255).toFixed(0)}, ${(g*255).toFixed(0)}, ${(b*255).toFixed(0)}, ${a})`
-        //console.log("RGB: ", UIColor, "Dominance: ",res[UIColor], "HEX:", hex);
-        swatches.push({color: rgba,
-                       population: res[UIColor],
-                       bodyTextColor: textColor,
-                       titleTextColor: textColor,
-                       swatchInfo: UIColor,
+        const r = uicolors[0];
+        const g = uicolors[1];
+        const b = uicolors[2];
+        const a = uicolors[3];
+        const hex = `#${toHex(r * 255)}${toHex(g * 255)}${toHex(b * 255)}${toHex(0.6)}`;
```

Comment:
Why the change from `toHex(a*255)` to `toHex(0.6)`?

Reply:
Because in my case,  I only need 60% alpha
I can make it as one of available in options, (called opacity)
What do you think?

Reply:
I think we should return the same thing for alpha as is returned by the native library (a*255). I don't think the extra complication of an option for opacity is necessary since it's trivial to modify the value returned by `react-native-palette`. Besides, it wouldn't support all use cases, for example, if rather than a fixed opacity (0.6) you wanted half of whatever opacity the library returned. Then you'd still have to modify the returned value.